crates/std_detect/src/detect/mod.rs

```diff
@@ -64,7 +64,10 @@ mod bit;
 mod cache;
 
 cfg_if! {
-    if #[cfg(any(target_arch = "x86", target_arch = "x86_64"))] {
+    if #[cfg(miri)] {
+        #[path = "os/other.rs"]
```

Comment:
Can you please add a comment here saying that on miri all features are disabled ?

Reply:
This comment should maybe add that this is correct because for features for which `cfg(target_feature)` is true, the run-time detection logic is not invoked, so `is_x86_feature_detected!` will return the same answer as miri for those anyways. 

cc @RalfJung

Reply:
I have no idea what this means. I have never used run-time feature detection so none of this code here means anything to me.